### PR TITLE
SPDX identifiers and a CI job to check them all

### PR DESCRIPTION
**Pull request by bagder**
- All files have prominent copyright and SPDX identifier
- If not embedded in the file, in the .reuse/dep5 file
- All used licenses are in LICENSES/ (not shipped in tarballs)
- A new REUSE CI job verify that all files are OK

### CODE REVIEW

File: LICENSES/curl.txt [hunk 1]
```diff
@@ -0,0 +1,22 @@
+COPYRIGHT AND PERMISSION NOTICE
+
+Copyright (C) Daniel Stenberg, <daniel@haxx.se>, and many
+contributors, see the THANKS file.
```
Comment (vszakats) [inline LICENSES/curl.txt:4]: The `THANKS` file is `docs/AUTHORS` in libssh2. Perhaps renaming it to `THANKS` would be the best?

Comment (vszakats) [reply LICENSES/curl.txt]: Me oops. I missed that this is the curl license. This is no biggie, or we might also change it to an URL.

..but, does libssh2 uses the curl license?

**Comment by bagder** [comment]
The final set of files with unclear license information:

* config.rpath (brought by e1bc0adcb1bc96625062094b108968a8eba9fd60)
* docs/INSTALL_AUTOTOOLS (from 8bb6cf7f956f, "This file is free documentation; the Free Software Foundation gives unlimited permission to copy, distribute and modify it.")
* m4/lib-ld.m4 (brought by e1bc0adcb1bc96625062094b108968a8eba9fd60)
* m4/lib-link.m4 (brought by e1bc0adcb1bc96625062094b108968a8eba9fd60)
* m4/lib-prefix.m4 (brought by e1bc0adcb1bc96625062094b108968a8eba9fd60)

**Comment by bagder** [comment]
They seem to be under the [FSFULLR](https://spdx.org/licenses/FSFULLR.html) license

**Comment by bagder** [comment]
Ping @mxmehl anything obvious you can think of I might have missed here?

File: src/agent_win.c [hunk 2]
```diff
@@ -35,6 +35,8 @@
  * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
  * USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY
  * OF SUCH DAMAGE.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
```
Comment (vszakats) [inline src/agent_win.c:39]: This file has 3 extra copyrights/licenses below the main copyright header.

File: tests/test_read_algos.test [hunk 3]
```diff
@@ -1,4 +1,6 @@
 #!/usr/bin/env bash
+# Copyright (C) The libssh2 project and its contributors.
+# SPDX-License-Identifier: BSD-3-Clause
 
 # Copyright (C) Viktor Szakats
```
Comment (vszakats) [inline tests/test_read_algos.test:5]: This line could replace the generic Copyright header in line 2.

File: tests/test_sshd.test [hunk 4]
```diff
@@ -1,7 +1,9 @@
 #!/usr/bin/env bash
-
+# Copyright (C) The libssh2 project and its contributors.
+# SPDX-License-Identifier: BSD-3-Clause
+#
 # Written by Simon Josefsson and Viktor Szakats
```
Comment (vszakats) [inline tests/test_sshd.test:5]: Same case as `Makefile.mk`: Maybe we could convert the "written by" line to a copyright header here? It means the same thing in practice, though IANAL.

File: src/agent_win.c (hunk 2 above)
Comment (bagder) [reply src/agent_win.c:39]: Indeed, will adapt.

File: Makefile.mk [hunk 5]
```diff
@@ -7,6 +7,8 @@
 #
 # Look for ' ?=' to find accepted customization variables.
 #
+# Copyright (C) The libssh2 project and its contributors.
+# SPDX-License-Identifier: BSD-3-Clause
```
[the remark this answers is not in the document]
Comment (bagder) [reply Makefile.mk:11]: :+1:

File: tests/test_read_algos.test (hunk 3 above)
Comment (bagder) [reply tests/test_read_algos.test:5]: :+1:

File: tests/test_sshd.test (hunk 4 above)
Comment (bagder) [reply tests/test_sshd.test:5]: :+1:

File: Makefile.mk [hunk 6]
```diff
@@ -3,10 +3,11 @@
 # Makefile for building libssh2 with GCC-like toolchains.
 # Use: make -f Makefile.mk [help|all|clean|dist|distclean|dyn|objclean|example|exampleclean|test|testclean]
 #
-# Written by Guenter Knauf and Viktor Szakats
+# Copyright (C) by Guenter Knauf and Viktor Szakats
```
Comment (vszakats) [inline Makefile.mk:6]:
```suggestion
# Copyright (C) Guenter Knauf
# Copyright (C) Viktor Szakats
```
To bring this to the layout used in other files.

File: tests/test_sshd.test [hunk 7]
```diff
@@ -1,7 +1,7 @@
 #!/usr/bin/env bash
-
-# Written by Simon Josefsson and Viktor Szakats
-
+# Copyright (C) Simon Josefsson and Viktor Szakats
```
Comment (vszakats) [inline tests/test_sshd.test:2]:
```suggestion
# Copyright (C) Simon Josefsson
# Copyright (C) Viktor Szakats
```

**Comment by mxmehl** [comment]
> Ping @mxmehl anything obvious you can think of I might have missed here?

Sorry, I was on vacation. But it looks like you did a great job. Thank you!

The cherry on the top would be to register the repository with the [REUSE API](https://api.reuse.software/projects) and include the resulting "compliant" badge e.g. in the README file. [curl is also registered](https://api.reuse.software/info/github.com/curl/curl) though I'm not sure whether it was @bagder who did it.

**Comment by bagder** [comment]
Thanks @mxmehl. I've registered this project there now and also @c-ares which was also made compliant a few days ago.

**Comment by bagder** [comment]
I'm not fan of badges, so none of these three projects get that. For the moment at least.

**Pull request merged**